falcon/stream.py

```diff
@@ -14,11 +14,18 @@
 
 """WSGI BoundedStream class."""
 
+from __future__ import annotations
 import io
+from typing import BinaryIO, Callable, List, Optional, TypeVar, Union
 
 __all__ = ['BoundedStream']
 
 
+Result = TypeVar('Result', bound=Union[bytes, List[bytes]])
+
+OptionalInt = Optional[int]
```

Comment:
Not crucial, but maybe we could move this `OptionalInt` (or `MaybeInt`) to `falcon/typing.py`?

Alternatively, we could maybe just inline since `OptionalInt` is only 2 characters shorter than `Optional[int]`, doesn't save much real estate on screen.

Reply:
Inlined.